src/Compilers/CSharp/Test/Symbol/Symbols/UnsignedRightShiftTests.cs

```diff
@@ -0,0 +1,1873 @@
+        [InlineData("nint", "byte", "(System.IntPtr.Size == 4 ? int.MinValue : long.MinValue)", "nint", "nuint")]
+        [InlineData("nint", "ushort", "(System.IntPtr.Size == 4 ? int.MinValue : long.MinValue)", "nint", "nuint")]
+        [InlineData("nuint", "char", "(System.IntPtr.Size == 4 ? uint.MaxValue : ulong.MaxValue)", "nuint", "nuint")]
+        [InlineData("nuint", "sbyte", "(System.IntPtr.Size == 4 ? uint.MaxValue : ulong.MaxValue)", "nuint", "nuint")]
+        [InlineData("nuint", "short", "(System.IntPtr.Size == 4 ? uint.MaxValue : ulong.MaxValue)", "nuint", "nuint")]
+        [InlineData("nuint", "int", "(System.IntPtr.Size == 4 ? uint.MaxValue : ulong.MaxValue)", "nuint", "nuint")]
+        [InlineData("nuint", "byte", "(System.IntPtr.Size == 4 ? uint.MaxValue : ulong.MaxValue)", "nuint", "nuint")]
+        [InlineData("nuint", "ushort", "(System.IntPtr.Size == 4 ? uint.MaxValue : ulong.MaxValue)", "nuint", "nuint")]
+        public void BuiltIn_01(string left, string right, string leftValue, string result, string unsignedResult)
+        {
+            var source1 =
+@"
+class C
+{
+    static void Main()
+    {
+        var x = (" + left + @")" + leftValue + @";
+        var y = (" + right + @")1;
+        var z1 = x >>> y;
+        var z2 = x >> y;
+
+        if (z1 == unchecked((" + result + @")(((" + unsignedResult + @")(" + result + @")x) >> y))) System.Console.WriteLine(""Passed 1"");
+
+        if (x > 0 ? z1 == z2 : z1 != z2) System.Console.WriteLine(""Passed 2"");
+
+        if (z1.GetType() == z2.GetType() && z1.GetType() == typeof(" + result + @")) System.Console.WriteLine(""Passed 3"");
+    }
+
+    " + result + @" Test1(" + left + @" x, " + right + @" y) => x >>> y; 
+    " + result + @" Test2(" + left + @" x, " + right + @" y) => x >> y; 
+}
+";
+            var compilation1 = CreateCompilation(source1, options: TestOptions.DebugExe,
+                                                 parseOptions: TestOptions.RegularPreview);
+
+            var verifier = CompileAndVerify(compilation1, expectedOutput: @"
+Passed 1
+Passed 2
+Passed 3
+").VerifyDiagnostics();
+
+            string actualIL = verifier.VisualizeIL("C.Test2");
+            verifier.VerifyIL("C.Test1", actualIL.Replace("shr.un", "shr").Replace("shr", "shr.un"));
+
+            var tree = compilation1.SyntaxTrees.Single();
+            var model = compilation1.GetSemanticModel(tree);
+            var unsignedShift = tree.GetRoot().DescendantNodes().OfType<BinaryExpressionSyntax>().Where(e => e.Kind() == SyntaxKind.UnsignedRightShiftExpression).First();
+            var shift = tree.GetRoot().DescendantNodes().OfType<BinaryExpressionSyntax>().Where(e => e.Kind() == SyntaxKind.RightShiftExpression).First();
+
+            Assert.Equal("x >>> y", unsignedShift.ToString());
+            Assert.Equal("x >> y", shift.ToString());
+
+            var unsignedShiftSymbol = (IMethodSymbol)model.GetSymbolInfo(unsignedShift).Symbol;
```

Comment:
Consider verifying `GetTypeInfo` too, or adding to test plan

Reply:
> Consider verifying GetTypeInfo too, or adding to test plan

I do not find this particularly interesting. Completeness is the only reason to have them. I am not adding new bound nodes, there is nothing new there in terms of calculating type info. Symbol info is interesting because we are supposed to synthesize operator symbols for intrinsic operators.